e2e/Metamask.ts

```diff
@@ -0,0 +1,287 @@
+
+    for (let i = 0; i < 12; i++) {
+      await signUpInputGroup.nth(i).fill(this.mnemonic[i]);
+    }
+
+    await this.registerPage
+      .getByRole('button', {
+        name: 'Confirm Secret Recovery Phrase',
+      })
+      .click();
+
+    await this.registerPage.waitForLoadState();
+
+    const passwordField = this.registerPage.getByRole('textbox').nth(0);
+    const confirmPasswordField = this.registerPage.getByRole('textbox').nth(1);
+
+    // password/registered account get lost when browser closes no need to include it in env
+    // private key must be secured
+    await passwordField.fill('e2e_test_account_password');
+    await confirmPasswordField.fill('e2e_test_account_password');
+
+    await this.registerPage.getByRole('checkbox').click();
+
+    await this.registerPage
+      .getByRole('button', { name: 'Import my wallet' })
+      .click();
+
+    await this.registerPage.getByRole('button', { name: 'Done' }).click();
+
+    await this.registerPage.getByRole('button', { name: 'Next' }).click();
+
+    await this.registerPage.getByRole('button', { name: 'Done' }).click();
+
+    await this.registerPage.close();
+  }
+
+  public async sign(opts: SigningOptions) {
+    const { txType, signingPopupPage, ctx } = opts;
+
+    if (txType !== TxType.EIP712 && txType !== TxType.EVM) {
+      throw new Error(
+        `Metamask doesn't support signing ${txType} transactions`,
+      );
+    }
+
+    if (txType === TxType.EVM) {
+      return await this.signEvm(signingPopupPage);
+    }
+
+    return await this.signEIP712(ctx, signingPopupPage);
+  }
+
+  private async signEvm(page: Page) {
+    const confirmBtn = page.getByRole('button', {
+      name: 'Confirm',
+    });
+
+    await expect(confirmBtn).toBeEnabled();
+    await confirmBtn.click();
+    await page.waitForEvent('close');
+  }
+
+  private async signEIP712(ctx: BrowserContext, page: Page) {
+    const signaturePage = await this.signEIP712StageOne(ctx, page);
+    await this.signEIP712StageTwo(signaturePage);
+  }
+
+  private async signEIP712StageOne(
+    ctx: BrowserContext,
+    page: Page,
+  ): Promise<Page> {
+    const signBtn = page.getByRole('button', {
+      name: 'Confirm',
+    });
+    await expect(signBtn).toBeEnabled();
+    const secondSigningPagePromise = ctx.waitForEvent('page');
+    await signBtn.click();
+    return await secondSigningPagePromise;
+  }
+
+  private async signEIP712StageTwo(page: Page) {
+    const signBtn = page.getByRole('button', {
+      name: 'Confirm',
+    });
+
+    await expect(signBtn).toBeEnabled();
+
+    await signBtn.click();
+
+    await page.waitForEvent('close');
+  }
+
+  public async switchNetwork() {
+    const metaMaskPage = await this.registerPage.context().newPage();
+    await metaMaskPage.goto(this.extensionLink + 'home.html');
+    await metaMaskPage.waitForSelector('[data-testid="network-display"]', {
+      timeout: 10000,
+    });
+
+    await metaMaskPage.getByTestId('network-display').click();
```

Comment:
Everything above this was directly imported from the webapp application.

Everything below is new since we don't have the "Switch Network" button that adjusts to our chain, so we have to manually register the evm chain, then switch to it, then add our existing wallet (which is currently one of the dev wallets with its key in local .env).